### PR TITLE
Fix Etag Mismatch auto saving

### PR DESCRIPTION
**Pull request by JillieBeanSim**
## Proposed changes

fix for https://github.com/zowe/vscode-extension-for-zowe/issues/2277

Remove duplicate code and updated the handling of dirty file when etag mismatch error for MVS and z/OS UNIX files by marking file unsaved before pulling the latest changes to the local file then use the vscode built in command `workspace.files.action.compareWithSaved`.

This PR was originally based against maintenance so will also bring in the commits for 2.8.2 fix & release

Fixed the error returned by FTP ext to trigger the diff editor with mismatch etag.

## Release Notes

Milestone: 2.9.0

Changelog: 
  ZE: Fixed an issue with mismatch etag error returned not triggering the diff editor and possible loss of data due to the issue. [#2277](https://github.com/zowe/vscode-extension-for-zowe/issues/2277)
  zFTP: - Fixed an issue with mismatch etag, correcting error message sent to Zowe Explorer to trigger diff editor. [#2277](https://github.com/zowe/vscode-extension-for-zowe/issues/2277)

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Further comments

Theia:
Screen Shot 2023-06-02 at 7 45 51 AM

zFTP:
Screen Shot 2023-06-02 at 8 26 27 AM

### CODE REVIEW

**Review by traeok — COMMENTED** [review]
LGTM, thanks @JillieBeanSim 🙂 just had a question regarding implementation for `putContents`

**Review by traeok — DISMISSED** [review]

**Review by rudyflores — APPROVED** [review]
Tested this and it seems to be handling the conflicts and displaying the diff view on VS Code to handle saving mismatches, thanks @JillieBeanSim !

**Review by traeok — APPROVED** [review]

File: packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpMvsApi.ts [hunk 1]
```diff
@@ -138,11 +138,9 @@ export class FtpMvsApi extends AbstractFtpApi implements ZoweExplorerApi.IMvs {
...
                 const contentsTag = await this.getContentsTag(dataSetName);
                 if (contentsTag && contentsTag !== options.etag) {
-                    // TODO: extension.ts should not check for zosmf errors.
-                    await Gui.errorMessage("Save conflict. Please pull the latest content from mainframe first.", {
-                        logger: ZoweLogger,
-                    });
-                    throw new Error();
+                    result.success = false;
+                    result.commandResponse = "Rest API failure with HTTP(S) status 412 Save conflict.";
+                    return result;
```
[the remark this answers is not in the document]
Comment (JillieBeanSim) [reply packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpMvsApi.ts:143]: Hey @traeok the FTP extension is just following the same protocol used with zosmf, I did notice USS and MVS are handled differently but that is how zowe cli was returning these and Zowe Explorer was built around that

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋 
now we get the compare button for both FTP and ZE 🥳

**Pull request merged**